### ISSUE TITLE
Relationships aren't created with nodes with specific label spellings

### ISSUE THREAD

**Issue by mattfiocca** [graphaware/neo4j-php-client#40]
This is totally weird, and I have no doubt this might be my setup, but take the following query:

	$result = $this->run("
		MATCH (s:BRIEF {id:{brief_id}})
		CREATE (n:BRIEFNOTECARD)
		SET n += {data}
		CREATE (n)-[:CARD_OF {order:0}]->(s)
		RETURN n
	", [ 'data' => $data, 'brief_id' => $brief_id ] );

Using the php client, I get a match on `BRIEF`, then create a new `BRIEFNOTECARD`, then attempt to create a relationship on those nodes, `:CARD_OF`. The match is working on `BRIEF`, and the `BRIEFNOTECARD` is being created, but the relationship edge is just not being created. 

Here's the weird part, when i change the spelling of the label on the new node to something like `BRIEFNOTECART` (just changing the last letter), everything works as expected.

ex:

	$result = $this->run("
		MATCH (s:BRIEF {id:{brief_id}})
		CREATE (n:BRIEFNOTECART)
		SET n += {data}
		CREATE (n)-[:CARD_OF {order:0}]->(s)
		RETURN n
	", [ 'data' => $data, 'brief_id' => $brief_id ] );

The new node is both created and attached to the `BRIEF`. I thought maybe it was something with the letter `D` at the end, and tried a different word that ended with that, and everything was created correctly. I also tried replacing a random letter like the F in BRIEF with an S to spell `BRIESNOTECARD`, and that also works. Its just anytime the literal `BRIEFNOTECARD` exactly is used as a label in my case, that the edge isn't created.

I'm losing my mind.

Also, using the neo web browser, the queries work just fine in all cases. My issue is specific to this php library. 

Strange, right?

**UPDATE**

Using the label `BRIEFCARD` also fails to produce an edge creation, where something like `BRIEFTHING` works..

**Comment by ikwattro** [comment]
This is indeed really really weird, and for sure not related to the client. The following test is passing : 

```php
<?php

namespace GraphAware\Neo4j\Client\Tests\Issues;

use GraphAware\Neo4j\Client\Tests\Integration\IntegrationTestCase;

/**
 * Class Issue40Test
 * @package GraphAware\Neo4j\Client\Tests\Issues
 *
 * @group issues
 */
class Issue40Test extends IntegrationTestCase
{
    public function testIssue()
    {
        $this->emptyDb();
        $this->client->run('CREATE (:BRIEF {id: 123})');

        $query = 'MATCH (s:BRIEF {id:{brief_id}})
    CREATE (n:BRIEFNOTECARD)
    SET n += {data}
    CREATE (n)-[:CARD_OF {order:0}]->(s)
    RETURN n';

        $parameters = [
            'brief_id' => 123,
            'data' => [
                'key_1' => 'test',
                'key_2' => 'other'
            ]
        ];

        $this->client->run($query, $parameters);
        $this->assertGraphExist('(n:BRIEF {id: 123})<-[:CARD_OF {order:0}]-(:BRIEFNOTECARD)');
    }

    private function assertGraphExist($pattern)
    {
        $q = sprintf('MATCH %s RETURN *', $pattern);
        $result = $this->client->run($q);

        $this->assertTrue(0 !== $result->size());
    }
}
```

Maybe you can share more about your code, even privately if you want.

**Comment by ikwattro** [comment]
@mattfiocca any update on this ?

**Comment by mattfiocca** [comment]
ill send some more context the evening, I'm currently away. 

--
Matt Fiocca
Snapdragon Studios, LLC.
(219) 307-4205
(219) 405-6723 cell

**Comment by mattfiocca** [comment]
@ikwattro I found the problem, sorry to bother. I have code later on in the runtime of our app was erroneously clearing these relationships, at times, and not others. 

this has nothing to do with the client.

**Comment by ikwattro** [comment]
Cool, glad it works now !

On 6 June 2016 at 23:48, Matt Fiocca <notifications@github.com> wrote:

> Closed #40 <https://github.com/graphaware/neo4j-php-client/issues/40>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/graphaware/neo4j-php-client/issues/40#event-683509827>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABKlecL0Gy3S03TO0S01p1k_VpR_4KTwks5qJJXLgaJpZM4It9-2>
> .
>



-- 
Christophe Willemsen | Graph Aware Limited

Phone: +44 (0) 333 444 7274 | Mobile: +32 (0) 489 687 208
christophe@graphaware.com | @graph_aware | www.graphaware.com
